pkg/storegateway/bucket_stores.go

```diff
@@ -73,6 +73,11 @@ type BucketStores struct {
 	storesErrorsMu sync.RWMutex
 	storesErrors   map[string]error
 
+	instanceTokenBucket *util.TokenBucket
```

Comment:
nit: The bucket_stores.go can have very minimal changes if all the code can be moved to the limiter.go with another layer of abstraction.